app/(public)/getProducts/page.tsx

```diff
@@ -0,0 +1,117 @@
+'use client'
+
+import { useRouter } from "next/navigation";
+import { useState } from "react";
+
+export default function getProducts() {
+
+    const router = useRouter();
+    const [productName, setProductName] = useState('');
+    const [itemCode, setItemCode] = useState('');
+    const [realItemCode, setRealItemCode] = useState('');
+    const [quantity, setQuantity] = useState(0);
+    const [products, setProducts] = useState([]);
+
+
+    async function handlePost(e: React.FormEvent<HTMLFormElement>) {
```

Comment:
The async functions seem to be repetitive. Perhaps you could create a component for these and just pass the necessary values like the route, method, body, etc.